### PR TITLE
feat(frontend): add delta join support

### PR DESCRIPTION
**Pull request by skyzh**
Signed-off-by: Alex Chi <iskyzh@gmail.com>

## What's changed and what's your intention?

As title. We use `set RW_FORCE_DELTA_JOIN` to enable delta join for now.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)

close https://github.com/singularity-data/risingwave/issues/1964

### CODE REVIEW

File: src/meta/src/stream/fragmenter.rs [hunk 1]
```diff
@@ -256,13 +256,16 @@ where
...
-                if ENABLE_DELTA_JOIN {
+            if hash_join_node.is_delta_join {
+                if hash_join_node.get_join_type()? == JoinType::Inner
+                    && hash_join_node.condition.is_none()
+                {
                     return self.build_delta_join(current_fragment, stream_node);
-                };
+                } else {
+                    panic!(
+                        "only inner join without non-equal condition is supported for delta joins"
```
Comment (fuyufjh) [inline src/meta/src/stream/fragmenter.rs:266]: how about falling back to hash join here?

**Review by fuyufjh — APPROVED** [review]
LGTM. Looking forward to `create index` :lark-cry:

File: src/meta/src/stream/fragmenter.rs (hunk 1 above)
Comment (skyzh) [reply src/meta/src/stream/fragmenter.rs:266]: I'm afraid some changes might cause delta join not used by default. So if the flag is on, all hash joins must be rewritten to delta join, to ensure the code path is covered in e2e.

File: e2e_test/v2/streaming_1node/join-3way.slt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+select v1, v3, v5 from mv1;
+----
+0 0 0
+1 1 1
+2 2 2
+3 3 3
+
+halt
+
+# TODO: drop MV is not supported yet for delta joins
```
Comment (fuyufjh) [inline e2e_test/v2/streaming_1node/join-3way.slt:38]: Any issue to track this?

Comment (skyzh) [reply e2e_test/v2/streaming_1node/join-3way.slt:38]: https://github.com/singularity-data/risingwave/issues/1901

**Pull request merged**